web/access_control.go

```diff
@@ -78,6 +78,54 @@ func (s *AutograderService) isEnrolled(userID, courseID uint64) bool {
 	})
 }
 
+// isValidSubmission returns true if submitter has active course enrollment or
+// if submitting group belongs to the given course
+func (s *AutograderService) isValidSubmissionRequest(submission *pb.SubmissionRequest) bool {
```

Comment:
It is a bit confusing to name this isValid... since we use that for field validation in ag package. Need to think of a more representative name; will think more about it...

Reply:
Agree, but currently out of ideas.